JSQMessagesViewController/Views/JSQMessagesCollectionView.h

```diff
@@ -73,6 +73,26 @@
 @property (strong, nonatomic) UIColor *typingIndicatorEllipsisColor;
 
 /**
+ *  The image to be used as an avatar on a user's typing indicator. If not set, no avatar image will be shown. Setting this image will hide the typing indicator message bubble and ellipsis image.
+ */
+@property (strong, nonatomic) UIImage *typingIndicatorAvatarImage;
+
+/**
+ *  The string to indicate that a user is typing. If not set, a generic message will be shown. Setting this string will hide the typing indicator message bubble and ellipsis image.
+ */
+@property (copy, nonatomic) NSString *typingIndicatorMessage;
```

Comment:
Right, so the image and message should change every time, based on who is typing. With this setup, you'd have to repeatedly do `self.typingIndicatorAvatarImage = img;`, etc. 

For each 'presence' message, you'll need to configure the footer view, so that's where this should happen. We should just pass these 2 params to the configure method, no need for properties.